### PR TITLE
Small include cleanup based on session during developer workshop

### PR DESCRIPTION
**Pull request by SimonHeybrock**
What the title says. The main improvement is that we now avoid including `NexusFile` *everywhere* (previously `V3D` and `MatrixWorkspace` required it, solved by moving code from `MatrixWorkspace` into `SaveNexusProcessed`).

**To test:**

Should be fine if it compiles. Just code review.

Internal change, no release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [x] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/API/src/FileBackedExperimentInfo.cpp [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 #include <sstream>
 
-#include <nexus/NeXusException.hpp>
 #include <nexus/NeXusFile.hpp>
+#include <nexus/NeXusException.hpp>
```
Comment (soininen) [inline Framework/API/src/FileBackedExperimentInfo.cpp]: Was there a reason to break the alphabetic ordering here? If this is a mistake, please have a look at `LoadMcStasNexus.cpp`, ``.

**Review by soininen — CHANGES_REQUESTED** [review]
Looks good to me, except in my alphabetic ordering, 'Exception' comes before 'File'.

File: Framework/API/src/FileBackedExperimentInfo.cpp (hunk 1 above)
Comment (SimonHeybrock) [inline Framework/API/src/FileBackedExperimentInfo.cpp]: Yes it does not compile if it is the otherway round (I don't know why). I believe in the cases you mentioned the `NeXusFile.hpp` must get included by some other header before.

Comment (soininen) [inline Framework/API/src/FileBackedExperimentInfo.cpp]: :man_facepalming: for non-self-contained headers.

**Review by soininen — APPROVED** [review]

**Pull request merged**